runtime/service/npud/core/DynamicLoader.cc

```diff
@@ -47,7 +47,14 @@ DynamicLoader::DynamicLoader(const char *file, int flags)
   _backend = std::shared_ptr<Backend>(alloc(), [dealloc](Backend *b) { dealloc(b); });
 }
 
-DynamicLoader::~DynamicLoader() { dlclose(_handle); }
+DynamicLoader::~DynamicLoader()
+{
+  // NOTE
+  // The _backend shared_ptr must be explicitly deleted before
+  // the dynamic library handle is released.
+  _backend.reset();
+  dlclose(_handle);
+}
```

Comment:
(optional) I think destructor is not necessary. If we change `_handle` from raw pointer to `std::unique_ptr` which has deleter for `dlclose()`. Anyway, this explicit code is okay.